### ISSUE TITLE
[@sapui5/types@1.115] Type signatures broken for sap.ui.define

### ISSUE THREAD

**Issue by BenReim** [SAP/ui5-typescript#403]
Starting with version 1.115, the type signatures for `sap.ui.define` seem to be broken and type errors are raised: 
```shell
No overload matches this call.
```

**Steps to reproduce:** 
1. Clone https://github.com/SAP-samples/ui5-cap-event-app.git
2. Checkout `js-with-typescript-support`
3. `npm i -D @sapui5/types@1.115.1`
4. Open `Registration.controller.js` and observe ts errors

**Comment by akudev** [comment]
Confirmed. Triggered by [this commit](https://github.com/SAP/openui5/commit/69da8838d1cfe03abd3ad385141fc0ae4f93148a) and probably caused by a bug in the JSDoc parser or type generator which changes the rest parameter syntax to one single parameter.

Before the change, the type definition said `Function`, now it says `(p1: any) => any` but it should say `(...p1: any) => any`.

Thanks!

**Comment by akudev** [comment]
Should be fixed for 1.116.
Looking into a 115 downport.

**Comment by akudev** [comment]
That last commit was the downport, so 1.115.2 also gets the fix.

**Comment by BenReim** [comment]
@akudev With the `@sapui5/types@1.117.0` the same issue seems to be back again

**Comment by petermuessig** [comment]
@codeworrior something to discuss - somehow this fix doesn't seem to be in "still"... 🤔

**Comment by petermuessig** [comment]
Somehow the change hasn't been merged into the master branch and thus was missing in 1.117.0. The fix will come with 1.118 and there is a planned patch for 1.117 in October. The `@types/openui5` from Definitely Typed have been fixed manually.
